### DISCUSSION TITLE
How do I add a sidecar to rabbitmq - Kuberenetes

### DISCUSSION THREAD

**Discussion by Ruchira-R** [rabbitmq/cluster-operator#1366] [Q&A] [answered]
Hiya,
     I have a working setup with rabbitmq cluster operator and rabbitmq. The [example ](https://www.rabbitmq.com/kubernetes/operator/using-operator.html#override) doesn't specify how to add additional key values needed by the sidecar. In my case, the sidecar is filebeat and the task I'm trying to tackle is to send rabbitmq logs to elastic.
While setting up the Rabbitmq Cluster Operator, I can add the sidecar, but I cannot add additional key value pairs (Like how do I specify the ELK Host - the Logstash host in this case).

Any examples would help me a lot, thanks.

**Comment by lukebakken** [comment]
This comment by @ChunyiLyu may be of help:

https://github.com/rabbitmq/cluster-operator/discussions/1359#discussioncomment-5947237

**Comment by Zerpet** [comment, marked as answer]
There is a side car example in our repo: https://github.com/rabbitmq/cluster-operator/tree/main/docs/examples/sidecar

Once you have the side car override, it's just a matter of adding the `env` section, as you would do for any StatefulSet.

You can also use `kubectl explain` to explore what does an object accept. Specifically to your case:

```
kubectl explain rmq.spec.override.statefulSet.spec.template.spec.containers
```
